client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( options => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		options,
```

Comment:
validating here is a perfect idea. @rads has been working a lot with the Clojure/spec family of validation methods and maybe he would have some valuable input here.

it's probably worth considering what we want to validate here. it may be that our biggest concern is validating that we received legitimate data from the API but go no further. maybe we'd rather let the reducers in Calypso make sure they don't store data that is invalid for "business rules" or "application logic"

at the most fine-grained level we could do things like force data typing. for example, there are plenty of places where we get numbers from the API in string form or dates in ISO8601 formats. the native way in JavaScript to store numbers and dates is with `Number` and `Date` and here we can translate. we might also do some restructuring of data. what might be less clear is if we would want to risk duplicating code and mismatching logic checks like, "is this number greater than X" or "is this date too old?"